fastchat/serve/inference.py

```diff
@@ -95,6 +105,44 @@ def load_model(model_path, device, num_gpus, max_gpu_memory="13GiB",
     return model, tokenizer
 
 
+def load_compress_model(model_path, device):
+
+    # partially load model
+    tokenizer = AutoTokenizer.from_pretrained(model_path, use_fast=False)
+    base_pattern = os.path.join(model_path, "pytorch_model-*.bin")
+    files = glob.glob(base_pattern)
+    
+    with init_empty_weights():
+        config = AutoConfig.from_pretrained(model_path, low_cpu_mem_usage=True, device_map='auto',torch_dtype=torch.float16)
+        model = AutoModelForCausalLM.from_config(config)
+        linear_weights = get_compressed_list(model)
+
+
+    compressed_state_dict = {}
+
+    for file in files:
+        tmp_state_dict = torch.load(file, map_location="cpu")
```

Comment:
Is `map_colcation="cpu"` required?

Reply:
IIUC,  we should avoid loading params to GPU memory directly. Since the compression part is done on GPU, loading directly will increase the peak memory on GPU. Annotating `map_location="cpu"` can prevent this.

Reply:
@andy-yang-1 I think `map_colcation="cpu"` is the default argument.